expression/builtin_cast.go

```diff
@@ -1937,7 +1937,24 @@ func WrapWithCastAsDecimal(ctx sessionctx.Context, expr Expression) Expression {
 	}
 	types.SetBinChsClnFlag(tp)
 	tp.Flag |= expr.GetType().Flag & mysql.UnsignedFlag
-	return BuildCastFunction(ctx, expr, tp)
+	castExpr := BuildCastFunction(ctx, expr, tp)
```

Comment:
Will `WrapWithCastAsDecimal` be used when user explicit cast a value to decimal using `cast(xx as decimal)`?

Reply:
No. WrapWithCastAsDecimal don't need the target field type, and only used by implicit cast.
Explicit cast will use BuildCastFunction directly.
And cast(xx as decimal) is equivalent to cast(xx as decimal(10,0))...